app/models/applicant.rb

```diff
@@ -22,4 +22,18 @@ def status_progress
       self.status = 'In Progress'
     end
   end
+
+  def approved?(applicant_id)
+    application_pets = PetApplicant.where(applicant_id: applicant_id)
+    application_pets.all? do |pet|
+        pet.status == 'approve'
+    end
+  end
+
+  def all_decided?(applicant_id)
+    application_pets = PetApplicant.where(applicant_id: applicant_id)
+    application_pets.all? do |pet|
+      pet.status.present?
```

Comment:
same here we would want to refactor this to not have to iterate through the application_pets but use Active Record on the collection instead. I'm going to psuedo code because off the top of my head I can't think of the pure AR way to do it but something like `application_pets.where('status IS NOT NULL')`. Basically you want to use the power of the DB to find your objects not ruby.